.github/workflows/run-pro-build-and-test.yml

```diff
@@ -0,0 +1,88 @@
+name: Run Pro build & tests
+
+on:
+  workflow_call:
+    inputs:
+      latestMergeSha:
+        description: 'Latest Merge sha of OSS PR'
+        required: false
+        default: ''
+        type: string
+      thisBranchName:
+        description: 'Branch Name of OSS PR'
+        required: false
+        default: ''
+        type: string
+  pull_request:
+    types: [labeled]
+
+
+jobs:
+  # build and test pro master branch using current branch to ensure nothing is being broken by this PR
+  should-run-pro-build:
+    name: Build & Test Pro
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'build&TestPro') || (github.event_name != 'pull_request' && github.event_name != 'pull_request_target')
```

Comment:
I dont think we would need `pull_request_target` . It would be needed if we need community PR's to run as well. If you want to run community PR's as well, then add authorize step to this , just like https://github.com/liquibase/liquibase/blob/a181c5fd044796b60d1ca015c3b8b72a4bc07bd8/.github/workflows/build-branch.yml#L27
and https://github.com/liquibase/liquibase/blob/a181c5fd044796b60d1ca015c3b8b72a4bc07bd8/.github/workflows/build-branch.yml#L35